EIPS/eip-6051.md

```diff
@@ -0,0 +1,469 @@
+
+If  `signerPubKey`  is provided or  `recipient`  contains `signature` data, the implementation MUST perform signature verification. Missing data or incorrect format etc. SHALL fail the call and result in empty return and optional error logs.
+
+`signerPubKey` could have been further signed by another key pair `(trusted, trustedPubKey)`, which is trusted by the Sender application. In that case, `signerPubKey` is appended with the corresponding signature data, which SHOULD be verified against `trustedPubKey`. See **Test Cases** for further clarification.
+
+The implementation shall then proceed to retrieve the private key `sk` corresponding to  `account`, and follow the **Core Algorithms** to encrypt it.
+
+The return data is a byte array which contains first the ephemeral sender public key `S` (compressed if applicable), then `cipher` including any authentication tag, that is, `S || cipher`.
+
+#### R3. Request for Recipient to unwrap and intake the private key
+
+```javascript
+request({
+	method: 'eth_intakePrivateKey',
+	params: [
+		version,
+		recipientPublicKey, //  no signature this time
+		oob,
+		salt,
+		data
+	],
+})
+// expected return value: account
+```
+
+This time  `recipientPublicKey`  is only the ephemeral public key `R` generated earlier in the recipient side, just for the implementation to retrieve the corresponding private key `r`.  `data`  is the return value from the call to encapsulate private key, which is `S || cipher`.
+
+When the encapsulated private key `sk` is decrypted successfully, the implementation can process it further according to the designated purposes. Some general security guidelines SHALL be followed, for example, do  *not*  log the value, do securely wipe it after use, etc.
+
+The return value is the corresponding Ethereum address for `sk`, or empty if any error.
+
+### Options and Parameters
+
+Available elliptic curves are:
+
+-   secp256k1 (mandatory)
+-   Curve25519
+
+Available authenticated encryption schemes are:
+
+-   AES-128-GCM (mandatory)
+-   AES-256-GCM
+-   Chacha20-Poly1305
+
+Version string is simply concatenation of elliptic curve and AE scheme, for example, secp256k1-AES-128-GCM. Above lists allows a combination of six different concrete schemes. Implementations are encouraged to implement curve related logic separately from authenticated encryption scheme to avoid duplication and to promote interoperability.
+
+Signature algorithms for each curve is:
+
+-   secp256k1 --> ECDSA
+-   Curve25519 --> Ed25519
+
+## Rationale
+
+A crucial difference of this [EIP-6051](./eip-6051.md) with [EIP-5630](./eip-5630.md) is that, as the purpose of key encapsulation is to transport private key securely, the public key from the key-recipient should be ephemeral, and mostly used only one-time. While in EIP-5630 settings, the public key of message recipient shall be stable for a while so that message senders can encrypt messages without key discovery every time.
```

Comment:
```suggestion
A critical difference of this [EIP-6051](./eip-6051.md) with [EIP-5630](./eip-5630.md) is that, as the purpose of key encapsulation is to transport a private key securely, the public key from the key-recipient should be ephemeral, and mostly used only one-time. While in EIP-5630 settings, the public key of the message recipient shall be stable for a while so that message senders can encrypt messages without key discovery every time.
```